Editor/Window/DeploymentScenariosInput.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using Editor.Window;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace AmazonGameLift.Editor
+{
+    internal class DeploymentScenariosInput : StatefulInput
+    {
+        private InputState _inputState;
+        private bool _enabled;
+        private DeploymentScenarios _deploymentScenarios;
+
+        private readonly VisualElement _container;
+        private readonly VisualElement _radio2Group;
+        private readonly VisualElement _radio3Group;
+        private readonly Button _showMoreScenariosButton;
+
+        public Action<DeploymentScenarios> OnValueChanged;
+
+        public static readonly Dictionary<int, DeploymentScenarios> ScenarioIndexMap = new()
+        {
+            { 1, DeploymentScenarios.SingleRegion },
+            { 3, DeploymentScenarios.SpotFleet },
+            { 4, DeploymentScenarios.FlexMatch },
+        };
+
+        public DeploymentScenariosInput(VisualElement container, DeploymentScenarios initialValue, bool enabled)
+        {
+            var uxml = Resources.Load<VisualTreeAsset>("EditorWindow/Components/DeploymentScenarios");
+            container.Add(uxml.Instantiate());
+
+            _container = container;
+            _inputState = initialValue == DeploymentScenarios.SingleRegion ? InputState.Initial : InputState.Expanded;
```

Comment:
This seems like an extra enum that isn't necessary since it's only being used in a single location. Could you please store the initialValue directly and add a switch case over that directly? 

In terms of naming initialValue seems a bit vague. Could you please rename it to `deploymentScenario`?

Reply:
Swapped to bool